src/main/kotlin/com/github/avrokotlin/avro4k/Avro.kt

```diff
@@ -86,14 +87,17 @@ open class AvroInputStreamBuilder<T>(
 class AvroDeserializerInputStreamBuilder<T>(
    private val deserializer: DeserializationStrategy<T>,
    private val avro : Avro,
+   private val namingStrategy: NamingStrategy,
    converter: (Any) -> T
 ) : AvroInputStreamBuilder<T>(converter) {
-   val defaultReadSchema : Schema by lazy { avro.schema(deserializer.descriptor) }
+   val defaultReadSchema : Schema by lazy { avro.schema(deserializer.descriptor, namingStrategy) }
```

Comment:
I actually wouldn't want the user to pass a NamingStrategy to the AvroDeserializerInputBuilder / AvroOutputStreamBuilder). 
How about having the namingStrategy as a property of the Avro instance? It would be an implementation detail of the Avro class that shouldn't be known to classes like AvroDeserializerInputBuilder.

Reply:
Yes that probably makes sense or I wonder wondering where the user should pass this information